### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by wernerdegroot**

### CODE REVIEW

File: How to release.md [hunk 1]
```diff
@@ -38,6 +42,26 @@ Commit name should be something like `Release v1.0.3`.
...
+If you get an error saying something like
+
+```
+gpg: signing failed: Inappropriate ioctl for device
+```
+
+first export `GPG_TTY`:
+
+```
+export GPG_TTY=$(tty)
```
Comment (jqno) [inline How to release.md:56]: I have that one too. So annoying.

File: README.md [hunk 2]
```diff
@@ -9,14 +9,24 @@ Java code generation for applicative functors, selective functors and more.
...
 ## Getting started
 
 Java 8 or higher is required.
 
-Add the following dependencies:
+Add the required dependencies:
+
+* [nl.wernerdegroot.applicatives.processor:1.0.3](https://mvnrepository.com/artifact/nl.wernerdegroot.applicatives/processor/1.0.3)  
+
+    Annotation processor. Only needed during compilation. Hook into `maven-compiler-plugin` or include as dependency with scope `provided`.
```
Comment (jqno) [inline README.md:24]: Does it matter which?

File: README.md [hunk 3]
```diff
@@ -9,14 +9,24 @@ Java code generation for applicative functors, selective functors and more.
...
+* [nl.wernerdegroot.applicatives.runtime:1.0.3](https://mvnrepository.com/artifact/nl.wernerdegroot.applicatives/runtime/1.0.3)
+
+    Required runtime dependencies (only a handful of classes, and no transitive dependencies).
```
Comment (jqno) [inline README.md:27]: Nice that you mention that there are no transitive dependencies.
Also, nice that there are no transitive dependencies 😄

File: README.md [hunk 4]
```diff
@@ -95,7 +126,9 @@ CompletableFuture<Person> futurePerson =
...
+The method [`thenCombine`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#thenCombine-java.util.concurrent.CompletionStage-java.util.function.BiFunction-) is a nifty function that combines the result of two `CompletableFuture`s (by using a `BiFunction` that you provide). In this case, the `BiFunction` combines a first name (`String`) and last name (`String`) into a `Person` object.
+
+Useful though `thenCombine` might be, if you want to combine more than two `CompletableFuture`s you are out of luck. Although it is not impossible to do so with the functions that the Java standard library provides, you will have to accept a lot of boilerplate code. The next section describes the problem of combining more than two `CompletableFuture`s, and shows you how to solve it using the Java standard library. The solution should leave you somewhat dissatisfied. The section will continue by showing you how you can use this library to reduce the boilerplate to a minimum.
```
Comment (jqno) [inline README.md:131]: Foreshadowing! I like this, now I know I need to have a little patience to get my question answered

File: README.md [hunk 5]
```diff
@@ -138,9 +171,32 @@ CompletableFuture<List<Move>> futureMoves = ...;
 
 How do you combine those?
 
-Unfortunately, `thenCombine` won't be of much help. It's capable of combining two `CompletableFuture`s, but not any more than that. 
+Unfortunately, `thenCombine` won't be of much help. It's capable of combining two `CompletableFuture`s, but not any more than that. Unfortunately, the authors of the Java standard library did not provide an overload for `thenCombine` that three or more `CompletableFuture`s.
```
Comment (jqno) [inline README.md:174]: The end of this sentence does not parse grammatically

File: README.md [hunk 6]
```diff
@@ -138,9 +171,32 @@ CompletableFuture<List<Move>> futureMoves = ...;
...
+                
+                // Create `PokemonCard`:
+                return new PokemonCard(name, hp, energyType, moves);
+        });
+
+// Implementation of `NameAndHp` and `NameAndHpAndEnergyType` 
+// not provided for the sake of brevity.
+```
+
+This is clearly no solution for a person who demands excellence from their programming language!
```
Comment (jqno) [inline README.md:197]: Haha, indeed! INDEED

File: README.md [hunk 7]
```diff
@@ -219,6 +275,8 @@ public class CompletableFutures implements CompletableFuturesOverloads {
 }
 ```
 
+Note that the `CompletableFutures` class as described above is already conveniently included for you in the `prelude` module. Check out the [implementation](https://github.com/wernerdegroot/applicatives/blob/main/prelude/src/main/java/nl/wernerdegroot/applicatives/prelude/CompletableFutures.java) and the [tests](https://github.com/wernerdegroot/applicatives/blob/main/prelude/src/test/java/nl/wernerdegroot/applicatives/prelude/CompletableFuturesTest.java).
```
Comment (jqno) [inline README.md:278]: 👍🏻👍🏻

File: README.md [hunk 8]
```diff
@@ -280,62 +338,64 @@ public class RandomGeneratorFunctions implements RandomGeneratorFunctionsOverloa
...
 
 ```java
 Function<Random, Person> randomPerson = 
         RandomGeneratorFunctions.instance().combine(
-            randomPersonId,
-            randomFirstName,
-            randomLastName,
-            randomBirthDate,
-            Person::new
+            randomName,
```
Comment (jqno) [inline README.md:346]: There's still a straggling reference to Person on line 344 (GitHub won't let me comment there)

File: README.md [hunk 9]
```diff
@@ -280,62 +338,64 @@ public class RandomGeneratorFunctions implements RandomGeneratorFunctionsOverloa
...
+        BiFunction<? super A, ? super B, ? extends C> combinator) {                                   
+                                                          ▲                                           
+                        ┌────────────────────┐            │                                           
+        return ...;  ◀──┤ This is up to you! │            │                                           
+                        └────────────────────┘            │                                           
+    }                                 ┌───────────────────┴───────────────────┐                       
+}                                     │    Combinator function is always a    │                       
+                                      │     BiFunction with contravariant     │                       
+                                      │ parameters and covariant return type. │                       
+                                      └───────────────────────────────────────┘
```
Comment (jqno) [inline README.md:398]: Still doesn't render correctly on my phone.

But screw my phone, as long as it renders on a desktop, it's fine. Nobody is going to use this library on a phone (I hope)

**Review by jqno — COMMENTED** [review]
Nice! Very nice.

File: README.md (hunk 8 above)
Comment (wernerdegroot) [reply README.md:344]: Great catch! Thanks!

**Pull request merged**